utilities/input_tools.py

```diff
@@ -241,7 +241,38 @@ def add_charge_axis(h, charge):
         hnew[...,charge_axis.index(charge)] = h.view(flow=True)
     return hnew
 
-def getPOInames(rtfile, poi_type="mu"):
+def getFitresult(fitresult_filename):
+    if fitresult_filename.endswith(".root"):
+        return uproot.open(fitresult_filename)
+    elif fitresult_filename.endswith(".hdf5"):
+        return h5py.File(fitresult_filename, mode='r')
+    else:
+        logger.warning(f"Unknown format of fitresult {fitresult}")
+        return None
+
+def getPOInames(fitresult_file, poi_type="mu"):
+    if isinstance(fitresult_file, h5py.File):
+        names = getPOInamesH5(fitresult_file, poi_type)
+    else:
+        names = getPOInamesRoot(fitresult_file, poi_type)
+
+    if len(names)==0:
+        logger.warning('No free parameters found (neither signal strenght(s), nor W mass)')
```

Comment:
Maybe should raise an exception here and catch it later?

Reply:
We still want to be able to run the script for fits w/o POI, for example to produce the pulls in the ptll fit. 
What is the benefit of having an exception and catching it? I don't know